src/mcp_proxy/__main__.py

```diff
@@ -108,7 +108,11 @@ def main() -> None:
         default=[],
         help="Allowed origins for the SSE server. Can be used multiple times. Default is no CORS allowed.",  # noqa: E501
     )
-
+    sse_server_group.add_argument(
+        "--auth-token",
+        default=None,
+        help="Authentication token for the SSE server. Default is no authentication.",
```

Comment:
There are a couple of existing conventions either the `API_ACCESS_TOKEN` env var or `--headers` and I believe these for going in the opposite direction. It might be helpful to explicitly clarify here the difference (e.g. explicitly state this is adding authentication to the proxy itself, and not passed to the downstream SSE server like the others. (Either clarifying those or this one or both)

Also suggest updating `README.md`

Reply:
This is adding header-based authentication to the proxy, yes. Specifying the expected "Bearer" value for the "Authorization" header. I'm certainly willing to change the naming or something if that would be better though.